_posts/en/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,566 @@
+was done---and they'd all need to use wallets compatible with such a
+child transaction signing protocol.  The non-interactive form with COSHV
+would only require them to submit a Bitcoin address and an email address
+(or some other protocol address for delivery of the committed outputs).
+
+### More advanced usage
+
+In our previous examples, getting the distribution transaction confirmed
+was an all-or-nothing endeavor.  Either all ten receivers finally
+received their money at the same time, or all of them were still waiting
+for that confirmation to occur.  However, with Taproot and COSHV,
+it's possible to allow each receiver to individually withdraw their
+money from the covenant while also allowing most the other receivers to
+keep their money offchain for the moment.
+
+For example, let's imagine Alice paid each of our ten receivers 1 BTC
+using a tree of outputs where each output paid two or three other outputs until
+an individual's money was withdrawn from the covenant.  A complete tree
+might for ten participants might look like this:
+
+![Example output tree for ten participants](/img/posts/2019-05-tx-tree-5.png)
+{:.center}
+
+If receiver Bob wants to individually withdraw now, he could
+follow the whole path until he received his output, allowing him to
+create only five outputs instead of the ten he'd need to create if
+everyone had to exit at the same time.
+
+![Example output tree to allow participant Bob to exit](/img/posts/2019-05-tx-tree-6.png)
+{:.center}
+
+Since Bob is the first participant to withdraw, he has to create more
+outputs than everyone else.
+Anyone else exiting after Bob would need to create even fewer outputs as
+they could just start spending from one of the intermediate COSHV
+outputs that Bob had caused to be confirmed onchain.  Some people (like Charlie in the above
+example) would get their exit outputs as a byproduct of another person's
+exit.  Overall, the total number of outputs that need to be created
+would be greater than the number of participants.  However, if nobody
+tried to exit early, a Taproot leaf at the top level could optionally
+allow everyone to exit together using a normal batched payment, which is
+just one output per receiver.
+
+Even with the simple example above we again see the key benefit
+of COSHV is its ability to allow Alice to construct this complex
+multi-transaction tree without any interaction from the receivers after
+she has each participant's own desired withdrawal address.  It's
+possible to construct the same tree using just the pre-signed
+transactions we saw in the first sub-section above, but each of the
+possible output sets would be a separate transaction that would need to
+be interactively signed by all participants before the setup transaction
+could be broadcast.  For larger numbers of participants, this could
+easily require the signing of many thousands of transactions by each
+participant.
```

Comment:
Also each participant only needs to verify their own path's commitments, other participants subtrees are not needed, which should improve privacy.